pkg/firewalld/firewalld.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2018 CNI authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package firewalld
+
+import (
+	"log"
+	"net"
+	"strings"
+
+	systemddbus "github.com/coreos/go-systemd/dbus"
+	"github.com/godbus/dbus"
+)
+
+const (
+	activeState = "active"
+	mode        = "replace"
+	unitName    = "firewalld.service"
+
+	firewalldName          = "org.fedoraproject.FirewallD1"
+	firewalldPath          = "/org/fedoraproject/FirewallD1"
+	firewalldZoneInterface = "org.fedoraproject.FirewallD1.zone"
+
+	errUnknownSource  = "UNKNOWN_SOURCE"
+	errZoneAlreadySet = "ZONE_ALREADY_SET"
+)
+
+// IsRunning checks whether firewalld service is active (running).
+func IsRunning() bool {
+	conn, err := systemddbus.New()
```

Comment:
I would actually punt the bus connection creation to the caller, and have the caller pass the connection object to this function.  Same below with AddSourceToZone() and RemoveSourceFromZone().  The reason is that you can use the same bus connection for multiple requests, there's no reason to open/close the connection continuously.

So Add/Remove would create the bus connection once at their start and then use that same connection in their respective for loops.